app/models/host/managed.rb

```diff
@@ -561,8 +561,9 @@ def hostgroup_inherited_attributes
 
   def apply_inherited_attributes(attributes, initialized = true)
     return nil unless attributes
-    #don't change the source to minimize side effects.
-    attributes = hash_clone(attributes).with_indifferent_access
+    #convert possible strong parameters to unsafe hash (filtering out unsafe items) and
+    #clone to minimize side effects
+    attributes = hash_clone(attributes.to_h).with_indifferent_access
```

Comment:
I am a bit concerned that `attributes` won't have `#to_h` method (if somebody passes a regular `Hash` instead of params).
I am not sure that it's valid use case anyway, so I won't block this PR if it won't be fixed.